### DISCUSSION TITLE
Problem with monthdata

### DISCUSSION THREAD

**Discussion by roelvdh** [SBFspot/SBFspot#593] [Q&A] [answered]
After 3 years of succesfully running sbfspot on my Raspberry Pi I now notice a problem with monthly data that has existed for a long time. The problem only became visible after uploading Old Data to Domoticz, whereby large chunks of monthly data turned out to be unexisting. 
I am running 2 SMA-systems. Monthly data of both systems are either available or missing in /home/pi/smadata/20yy at the same pace, so the problem seems to relate to the execution of monthdata. The crontab is correct for monthdata and was never changed.
Over the last 3 years all data for /home/pi/smadata/20yy/nameX-20yymmdd.csv as well as for /home/pi/smadata/20yy/nameX-Spot-20yymmdd.csv are available and are correct.
Files /home/pi/smadata/20yy/nameX-20yymm.csv are missing for 8 months of 2021 and 7 months of 2022. The last correct file is for July 2022, containing 22-07-2022 as the last info. This is about the same date I updated sbfspot, so possibly something has gone wrong with the update.
My 2 questions:
1. How to solve this problem (re-installing sbfspot ?)
2. Is there a way to run monthdata for the missing months?

Edit:
I ran /usr/local/bin/sbfspot.3/monthdata manually. It takes 3 minutes on my RasPi and does not produce the 2 /home/pi/smadata/2022/nameX-202212.csv files I would expect

**Comment by sbf-** [comment, marked as answer]
> Is there a way to run monthdata for the missing months?

Running monthdata is only for the current month. Not sure if adding -am6 will overrule the -am1 in the monthdata script
Better is to run SBFspot manually in verbose mode and look for errors:
`/usr/local/bin/sbfspot.3/SBFspot -finq -v5 -ad0 -ae0 -sp0 -am6`
`-am6` means 6 archive months starting from december 2022
By adding `-startdate:20211201` you will request the months for 2021 (december, november, ...)

Run `SBFspot -?` for all possible arguments

**Reply by roelvdh** [reply]
Thank you, that is a great answer! I will try it tomorrow when the inverters are online.

Found the problem, a missing character in the $log variable
